src/virtual_dom/update_el.rs

```diff
@@ -111,3 +143,58 @@ where
         self.for_each(|item| item.update(el));
     }
 }
+
+// impl<Ms, I, U, P> UpdateEl<El<Ms>> for std::iter::Filter<I, P>
+// where
+//     I: Iterator,
+//     U: UpdateEl<El<Ms>>,
+//     P: FnMut(&I::Item) -> bool,
+// {
+//     fn update(self, el: &mut El<Ms>) {
+//         self.for_each(|item| item.update(el));
+//     }
+// }
+
+//impl<Ms, I, U, F> UpdateEl<El<Ms>> for std::iter::Map<I, F>
+//where
+//    I: Iterator,
+//    U: UpdateEl<Attrs>,
+//    F: FnMut(I::Item) -> U,
+//{
+//    fn update(self, el: &mut El<Ms>) {
+//        self.for_each(|item| item.update(el));
+//    }
+//}
+//
+//impl<Ms, I, U, F> UpdateEl<El<Ms>> for std::iter::Map<I, F>
+//where
+//    I: Iterator,
+//    U: UpdateEl<&Attrs>,
+//    F: FnMut(I::Item) -> U,
+//{
+//    fn update(self, el: &mut El<Ms>) {
+//        self.for_each(|item| item.update(el));
+//    }
+//}
+//
+//impl<Ms, I, U, F> UpdateEl<El<Ms>> for std::iter::Map<I, F>
+//where
+//    I: Iterator,
+//    U: UpdateEl<Style>,
+//    F: FnMut(I::Item) -> U,
+//{
+//    fn update(self, el: &mut El<Ms>) {
+//        self.for_each(|item| item.update(el));
+//    }
+//}
+//
+//impl<Ms, I, U, F> UpdateEl<El<Ms>> for std::iter::Map<I, F>
+//where
+//    I: Iterator,
+//    U: UpdateEl<&Style>,
+//    F: FnMut(I::Item) -> U,
+//{
+//    fn update(self, el: &mut El<Ms>) {
+//        self.for_each(|item| item.update(el));
+//    }
+//}
```

Comment:
?

Reply:
A failed attempt to make `update` work on `Filter`, like we have with `Map`, ie so you don't have to add `collect()`. I think it'd be nice to have, but can't get it working.

Reply:
https://github.com/seed-rs/seed/issues/128#issuecomment-570337605